packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/FVPVideoPlayerPlugin.m

```diff
@@ -406,6 +416,30 @@ - (void)updatePlayingState {
   _displayLink.running = _isPlaying || self.waitingForFrame;
 }
 
+- (void)updateRate {
+  // See https://developer.apple.com/library/archive/qa/qa1772/_index.html for an explanation of
+  // these checks.
+  // If status is not AVPlayerItemStatusReadyToPlay then both canPlayFastForward
+  // and canPlaySlowForward are always false and it is unknown whether video can
+  // be played at these speeds, updatePlayingState will be called again when
+  // status changes to AVPlayerItemStatusReadyToPlay.
+  float speed = _playbackSpeed.floatValue;
+  bool readyToPlay = _player.currentItem.status == AVPlayerItemStatusReadyToPlay;
+  if (speed > 2.0 && !_player.currentItem.canPlayFastForward) {
+    if (!readyToPlay) {
```

Comment:
> What if readyToPlay is false, and we set the speed to a value between 1.0 and 2.0?

Everything is fine then, the rate can be set anytime.

> Why are we allowing setting the speed for this value but not setting the speed for clamped value?

To avoid setting it to a value which was not requested and seems each change to rate triggers re-buffering and `playbackLikelyToKeepUp` as is stated in description.

> I'm still confused why this check can't be moved to the top of the function?

I chose to do it strictly only when needed. So you probably argue to do it always due to simplicity maybe. But I believe setting up the rate before `ReadyToPlay` may avoid some re-buffering as mentioned above.

Reply:
> Why are we allowing setting the speed for this value but not setting the speed for clamped value?

The problem isn't that we can't set the speed while buffering, it's that we don't know whether we need to clamp while buffering. But for 1.0-2.0 it's irrelevant because clamping wouldn't apply anyway.

(There is an unfortunate side-effect of this PR in that setting at unsupported value now silently clamps instead of throwing an exception, adding another potential way for native and Dart to get out of sync about state since Dart will think it's successfully set the out-of-range value, but we can address that edge case later with more event channel communication.)